### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by LeonCrashCode** [clab/dynet#762]
I pull the newest version of dynet 2.0. After using the function "set_dropout(pdrop)" for VanillaLSTMBuilder, there will be segmentation fault errors during running.

**Comment by pmichel31415** [comment]
Can you provide a minimal example? I couldn't reproduce the issue, here's what I tried (on the latest commit on master):

    import dynet as dy
    pc=dy.ParameterCollection()
    lstm = dy.VanillaLSTMBuilder(1,10,10,pc)
    lstm.set_dropout(0.5)
    trainer = dy.SimpleSGDTrainer(pc)
    for i in range(10):
        dy.renew_cg()
        lstm.set_dropout(0.5)
        x=[dy.inputTensor(np.random.uniform(size=10)) for t in range(5)]
        s=lstm.initial_state()
        y = s.transduce(x)[-1]
        l = dy.squared_norm(y)
        print(l.value())
        l.backward()
        trainer.update()

**Comment by neubig** [comment]
I'm closing this for lack of activity, but can re-open if we can get a minimal example.

**Comment by LeonCrashCode** [comment]
Thanks for your example. I reviewed my codes again, and find the reason why I get segment fault by applying dropout to VanillaLSTM. Actually, It is not a real bug, but requires regular coding.

If you build LSTM like this, you will get segmentation fault.
`lstm.start_new_sequence();
 lstm.set_dropout(0.5);`

In dynet, the dropout masks are initialized in the function start_new_sequence_imp(). So I change to make set_dropout before start_new_sequence, everything is ok.

**Comment by neubig** [comment]
OK, thanks! We'll try to fix to prevent it from throwing a segfault at least.
